### PR TITLE
[REV-41] Add GraphQL Queries and Types

### PR DESCRIPTION
**Pull request by tiagocardoso98**
Closes issue #41 
Added basic queries for all the database models

### CODE REVIEW

**Review by miguelrodrigues22 — APPROVED** [review]

**Review by JdFSilva — CHANGES_REQUESTED** [review]

File: photo/schema.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        if user_email or contest:
+            if user_email:
+                return ContestSubmission.objects.filter(picture__user__email=user_email)
+            return ContestSubmission.objects.filter(contest__id=contest)
+        return ContestSubmission.objects.all()
+
+
+@strawberry.type
+class Mutation:
+    pass
```
[the remark this answers is not in the document]
Comment (tiagocardoso98) [reply photo/schema.py:88]: Moved to the new file

File: photo/schema.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    CollectionType,
+    ContestSubmissionType,
+    ContestType,
+    PictureCommentType,
+    PictureType,
+    UserType,
+)
+
+
+# Query = strawberry_django.queries(User,  Picture, PictureComment, Collection, Contest, ContestSubmission, types=types)
```
[the remark this answers is not in the document]
Comment (tiagocardoso98) [reply photo/schema.py:24]: Removed

File: photo/schema.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    def contest_submissions(
+        self, user_email: str = None, id: int = None, contest: int = None
+    ) -> List[ContestSubmissionType]:
+        if id:
+            return ContestSubmission.objects.filter(id=id)
+        if user_email or contest:
+            if user_email:
+                return ContestSubmission.objects.filter(picture__user__email=user_email)
+            return ContestSubmission.objects.filter(contest__id=contest)
+        return ContestSubmission.objects.all()
```
[the remark this answers is not in the document]
Comment (tiagocardoso98) [reply photo/schema.py:83]: Moved to the new file

File: tests/test_queries/test_collection.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
+from django.test import TestCase
+
+from photo.schema import schema
+from tests.factories import CollectionFactory, PictureFactory, UserFactory
+
+
+class PictureTest(TestCase):
```
[the remark this answers is not in the document]
Comment (tiagocardoso98) [reply tests/test_queries/test_collection.py:13]: Fixed test name

File: tests/test_queries/test_collection.py [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+                        collections(user_email: $user_email) {
+                            name
+                            user {
+                                email
+                                name_first
+                                name_last
+                            }
+                            pictures {
+                                picture_path
+                            }
```
[the remark this answers is not in the document]
Comment (tiagocardoso98) [reply tests/test_queries/test_collection.py]: Removed extra fields

File: tests/test_queries/test_collection.py [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+class PictureTest(TestCase):
+    def setUp(self):
+        self.batch = 10
+        self.newPictures = PictureFactory.create_batch(
+            self.batch, user=UserFactory(user_profile_picture=True)
+        )
+        self.newColletions = CollectionFactory.create_batch(
+            self.batch, collection_pictures=self.newPictures
+        )
```
[the remark this answers is not in the document]
Comment (tiagocardoso98) [reply tests/test_queries/test_collection.py:21]: I use the others to guarantee that only the one specific object is being queried

File: revent/urls.py [hunk 7]
```diff
@@ -1,11 +1,11 @@
...
 
-from photo import views
+from photo.schema import schema
 
 urlpatterns = [
     path("admin/", admin.site.urls),
-    path("", views.hello, name="hello")
-    # In case we wanna have multiple url files for each app
-    # path('photo/', include('photo.urls'))
+    path("graphql/", csrf_exempt(GraphQLView.as_view(schema=schema))),
```
Comment (JdFSilva) [inline revent/urls.py:10]: please go back to using `AsyncGraphQLView`

**Review by JdFSilva — CHANGES_REQUESTED** [review]

**Review by JdFSilva — APPROVED** [review]

File: revent/urls.py (hunk 7 above)
Comment (tiagocardoso98) [reply revent/urls.py:10]: Theres is currently an error with the queries when using AsyncGraphQLView

**Pull request merged**